genomepy/annotation.py

```diff
@@ -0,0 +1,395 @@
+
+    def _contig_conversion_dict(self, matching_element: int):
+        """
+        build a conversion dict
+        returns the dict and a list of duplicate contig names (should be empty!)
+        """
+        conversion_table = {}
+        duplicate_contigs = []
+        with _open(self.genome_file, "r") as fa:
+            for line in fa:
+                if line.startswith(">"):
+                    line = line.strip(">\n").split(" ")
+                    if line[matching_element] in conversion_table:
+                        duplicate_contigs.append(line[matching_element])
+                    conversion_table[line[matching_element]] = line[0]
+        return conversion_table, list(set(duplicate_contigs))
+
+    def _conform_gtf(
+        self, conversion_dict: dict, filter_contigs: Optional[bool] = True
+    ):
+        """
+        Create an annotation gtf file with contigs conforming to the genome
+        Overwrites the existing gtf file.
+        Returns a list of contigs not present in the genome
+        """
+        missing_contigs = []
+        tmp_dir = mkdtemp(dir=self.genome_dir)
+        new_gtf_file = os.path.join(tmp_dir, os.path.basename(self.annotation_gtf_file))
+        with _open(self.annotation_gtf_file, "r") as old, _open(
+            new_gtf_file, "w"
+        ) as new:
+            for line in old:
+                splitline = line.split("\t")
+                if splitline[0] in conversion_dict:
+                    splitline[0] = conversion_dict[splitline[0]]
+                    line = "\t".join(splitline)
+                    new.write(line)
+                else:
+                    missing_contigs.append(splitline[0])
+                    if not filter_contigs:
+                        new.write(line)
+
+        os.replace(new_gtf_file, self.annotation_gtf_file)
+        rm_rf(tmp_dir)
+        return list(set(missing_contigs))
+
+    def bed_from_gtf(self):
+        """
+        Create an annotation bed file from an annotation gtf file.
+        Overwrites the existing bed file.
+        """
+        tmp_dir = mkdtemp(dir=self.genome_dir)
+
+        # unzip if needed
+        self.annotation_gtf_file, is_unzipped = gunzip_and_name(
+            self.annotation_gtf_file
+        )
+        new_bed_file = os.path.join(
+            tmp_dir, os.path.basename(re.sub(r"\.gz$", "", self.annotation_bed_file))
+        )
+
+        cmd = "gtfToGenePred {0} /dev/stdout | genePredToBed /dev/stdin {1}"
+        sp.check_call(cmd.format(self.annotation_gtf_file, new_bed_file), shell=True)
+
+        # gzip if needed
+        self.annotation_gtf_file = gzip_and_name(self.annotation_gtf_file, is_unzipped)
+        new_bed_file = gzip_and_name(
+            new_bed_file, self.annotation_bed_file.endswith(".gz")
+        )
+
+        os.replace(new_bed_file, self.annotation_bed_file)
+        rm_rf(tmp_dir)
+
+    def sanitize(self, filter_contigs: Optional[bool] = True):
+        """
```

Comment:
Function can be simplified by splitting it out more, for me it is hard to follow now. It should be three steps, correct?
1) sanitize if necessary
2) remove annotation not in reference genome
3) update README

However, this logic is mixed now, with three returns in the function